vlog4j-core/src/test/java/org/semanticweb/vlog4j/core/reasoner/implementation/LoadDataFromRdfTest.java

```diff
@@ -0,0 +1,134 @@
+				Expressions.makeVariable("p"), Expressions.makeVariable("o"));
+
+		for (final String rdfFileName : Arrays.asList("empty.nt", "empty.nt.gz")) {
+			final File emptyRdfFile = new File(FileDataSourceUtils.INPUT_FOLDER + rdfFileName);
+			final FileDataSource emptyDataSource = new RdfFileDataSource(emptyRdfFile);
+
+			try (final Reasoner reasoner = Reasoner.getInstance()) {
+				reasoner.addFactsFromDataSource(predicate, emptyDataSource);
+				reasoner.load();
+				reasoner.setAlgorithm(Algorithm.RESTRICTED_CHASE);
+				reasoner.reason();
+
+				try (final QueryResultIterator answerQuery = reasoner.answerQuery(queryAtom, true)) {
+					assertFalse(answerQuery.hasNext());
+				}
+				try (final QueryResultIterator answerQuery = reasoner.answerQuery(queryAtom, false)) {
+					assertFalse(answerQuery.hasNext());
+				}
+
+				reasoner.resetReasoner();
+				reasoner.load();
+				reasoner.setAlgorithm(Algorithm.SKOLEM_CHASE);
+				reasoner.reason();
+
+				try (final QueryResultIterator answerQuery = reasoner.answerQuery(queryAtom, true)) {
+					assertFalse(answerQuery.hasNext());
+				}
+				try (final QueryResultIterator answerQuery = reasoner.answerQuery(queryAtom, false)) {
+					assertFalse(answerQuery.hasNext());
+				}
+			}
+		}
+	}
+
+	@Test
+	public void testLoadTernaryFactsFromRdfFile() throws ReasonerStateException, EdbIdbSeparationException,
+	EDBConfigurationException, IOException, IncompatiblePredicateArityException {
+		final Predicate predicate = Expressions.makePredicate(ternaryPredicateName, 3);
+
+		for (final String rdfFileName : Arrays.asList(FileDataSourceUtils.unzippedNtFileRoot + ".nt",
+				FileDataSourceUtils.zippedNtFileRoot + ".nt.gz")) {
+			final File rdfFile = new File(FileDataSourceUtils.INPUT_FOLDER + rdfFileName);
+			final FileDataSource fileDataSource = new RdfFileDataSource(rdfFile);
+
+			try (final Reasoner reasoner = Reasoner.getInstance()) {
+				reasoner.addFactsFromDataSource(predicate, fileDataSource);
+				reasoner.load();
+
+				final QueryResultIterator queryResultIterator = reasoner.answerQuery(Expressions.makeAtom(predicate,
+						Expressions.makeVariable("s"), Expressions.makeVariable("p"), Expressions.makeVariable("o")),
+						true);
+				final Set<List<Term>> queryResult = QueryResultsUtils.collectQueryResults(queryResultIterator);
+
+				assertEquals(expectedTernaryQueryResult, queryResult);
+			}
+		}
+	}
+
+	@Test(expected = IOException.class)
+	public void testRdfFileNotOnDisk()
+			throws IOException, ReasonerStateException, EdbIdbSeparationException, IncompatiblePredicateArityException {
+		final File unexistingFile = new File("unexistingFile.nt");
+		assertFalse(unexistingFile.exists());
+		final FileDataSource fileDataSource = new RdfFileDataSource(unexistingFile);
+
+		try (final Reasoner reasoner = Reasoner.getInstance()) {
+			reasoner.addFactsFromDataSource(Expressions.makePredicate(ternaryPredicateName, 3), fileDataSource);
+			reasoner.load();
+		}
+	}
+
+}
```

Comment:
I see the need for this test. However, there is a problem: Because N-Triples enforces a predicate arity of 3 by its syntax, `toVLogPredicate()` returns a predicate of arity -1 on files with arities other than 3 (erroneous syntax). Therefore, `validateDataSourcePredicateArities()`only outputs a warning and never throws an `IncompatiblePredicateArityException`, because of the `if`-`else` structure. I can add the test, but it will fail until we add new behaviour.